### PR TITLE
Add gas tester

### PR DESCRIPTION
**Pull request by MTCam**
Adds `gas-in-box.py` example that is very convenient for testing code and model features.   Also adds some thermochemical mechanisms useful for testing the effect of increasing nspecies, and changing mechanisms.   

**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

**Comment by tulioricci** [comment]
Would it be easy/possible/useful to output DAG information in this driver?

**Comment by MTCam** [comment]
> Would it be easy/possible/useful to output DAG information in this driver?


I use it for that, and for testing lots of different stuff like free-stream preservation with TPEs.  It is essentiallly a more configurable `combozzle`.

File: examples/gas-in-box.py [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+    boundaries = {}
+    if not periodic_mesh:
+        if multiple_boundaries:
+            for idir in range(dim):
+                boundaries[BoundaryDomainTag(f"+{idir+1}")] = wall_bc
+                boundaries[BoundaryDomainTag(f"-{idir+1}")] = wall_bc
+        else:
+            boundaries = {BTAG_ALL: wall_bc}
+
+    acoustic_pulse = AcousticPulse(dim=dim, amplitude=100., width=.1, center=orig)
```
Comment (tulioricci) [inline examples/gas-in-box.py:543]: Just move this line to the "if add_pulse:" below.

File: examples/gas-in-box.py [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+    static_temp = 2076.43
+    # steepness of the smoothed function
+    theta_sc = 100
+    # cutoff, smoothness below this value is ignored
+    beta_sc = 0.01
+    gamma_sc = 1.5
+    alpha_sc = 0.3
+    kappa_sc = 0.5
+    s0_sc = -5.0
+    s0_sc = np.log10(1.0e-4 / np.power(order, 4))
```
Comment (tulioricci) [inline examples/gas-in-box.py:360]: This variable is duplicated

File: examples/gas-in-box.py [hunk 3]
```diff
@@ -0,0 +1,834 @@
...
+
+        logmgr.add_watches([
+            ("step.max", "step = {value}, "),
+            ("t_sim.max", "sim time: {value:1.6e} s\n"),
+            ("t_step.max", "------- step walltime: {value:6g} s, "),
+            ("t_log.max", "log walltime: {value:6g} s")
+        ])
+
+    velocity = np.zeros(shape=(dim,))
+    species_diffusivity = None
```
Comment (tulioricci) [inline examples/gas-in-box.py:248]: Move these 3 lines down to keep all transport stuff together.

File: examples/gas-in-box.py [hunk 4]
```diff
@@ -0,0 +1,834 @@
...
+
+        current_gas_state = mfs_compiled(current_cv, temperature_seed)
+
+    if logmgr:
+        from mirgecom.logging_quantities import logmgr_set_time
+        logmgr_set_time(logmgr, current_step, current_t)
+
+    visualizer = make_visualizer(dcoll)
+
+    initname = "pulse"
```
Comment (tulioricci) [inline examples/gas-in-box.py:588]: "gas-in-box"

**Review by tulioricci — APPROVED** [review]
Just left some minor comments. Merge at will.

File: examples/gas-in-box.py (hunk 3 above)
Comment (MTCam) [reply examples/gas-in-box.py:248]: I just need the velocity to stay put.

**Pull request merged**